creator-node/scripts/clock-data-migration.js

```diff
@@ -60,41 +64,119 @@ const populateClockVals = async () => {
     const models = await initDB(dbUrl)
 
     for (const { userId, secondaries } of nodeToUsersMap[node]) {
-      console.log(node, dbUrl, userId, secondaries)
-
-      const transaction = null // await models.sequelize.transaction()
-
-      const audiusUsers = await models.AudiusUser.findAll({
-        where: { blockchainId: userId },
-        transaction
-      })
-      console.log(audiusUsers)
-
-      if (!audiusUsers) continue
-
-      const cnodeUserUUID = audiusUsers[0].cnodeUserUUID
-
-      const cnodeUsers = await models.CNodeUser.findAll({
-        where: { cnodeUserUUID },
-        transaction
-      })
-      const tracks = await models.Track.findAll({
-        where: { cnodeUserUUID },
-        transaction
-      })
-      const files = await models.File.findAll({
-        where: { cnodeUserUUID },
-        transaction
-      })
-
-      // order all rows from Files, Tracks, AudiusUsers by "updatedAt" asc
-
-      // if dupe timestamps -> log out (this should never happen)
-
-      // await transaction.commit()
+      const transaction = await models.sequelize.transaction()
+
+      try {
+        // console.log('\n\n\n')
+        const audiusUsers = await models.AudiusUser.findAll({
+          where: { blockchainId: userId },
+          transaction
+        })
+
+        let cnodeUserUUID
+        if (audiusUsers && audiusUsers.length > 0) {
+          cnodeUserUUID = audiusUsers[0].cnodeUserUUID
+  
+          const tracks = await models.Track.findAll({
+            where: { cnodeUserUUID },
+            transaction
+          })
+          const files = await models.File.findAll({
+            where: { cnodeUserUUID },
+            transaction
+          })
+  
+          // Aggregate all data in array
+          const data = []
+          for (const audiusUser of audiusUsers) data.push([audiusUser, 'audiusUser'])
+          for (const track of tracks) data.push([track, 'track'])
+          for (const file of files) data.push([file, `file - ${file.dataValues.type}`])
+          // for (const datum of data) {
+          //   const entry = datum[0].dataValues
+          //   // console.log(`SIDTEST DATA ENTRY: ${entry.createdAt} - ${entry.clock}`)
+          // }
+  
+          // order all rows from Files, Tracks, AudiusUsers by "created" ASC (compares times in milliseconds)
+          data.sort((a, b) => {
+            const dA = new Date(a[0].dataValues.createdAt).getTime()
+            const dB = new Date(b[0].dataValues.createdAt).getTime()
+            const diff = dA - dB
+            return diff
+          })
+          // for (const datum of data) {
+          //   const entry = datum[0].dataValues
+          //   // console.log(`SIDTESTSORTED DATA ENTRY: ${entry.createdAt} - ${entry.clock} - ${datum[1]}`)
+          // }
+  
+          // Update each data table row with new clock2 value
+          let clockCounter = 0
```

Comment:
how can we accurately reject if the operation has already been run?

Reply:
just need to check if cnodeUser.clock is already > 0. this case shouldn't ever arise but v easy to add this check and we should def have it. adding now